src/lib/util/format-base-uri.ts

```diff
@@ -1,14 +1,14 @@
-export const formatBaseUri = (input: string): string => {
+export const formatBaseUri = (input?: string): string => {
     if (!input) return '';
     const firstChar = input[0];
     const lastChar = input[input.length - 1];
 
     if (firstChar === '/' && lastChar === '/') {
-        return input.substr(0, input.length - 1);
+        return input.substring(0, input.length - 1);
```

Comment:
`substr` is deprecated, following advice from: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr